pandas/tseries/timedeltas.py

```diff
@@ -37,7 +37,7 @@ def to_timedelta(arg, unit='ns', box=True, errors='raise', coerce=None):
 
     def _convert_listlike(arg, box, unit, name=None):
 
-        if isinstance(arg, (list,tuple)) or ((hasattr(arg,'__iter__') and not hasattr(arg,'dtype'))):
+        if isinstance(arg, (list, tuple)) or not hasattr(arg, 'dtype'):
```

Comment:
note that there's no need to check `hasattr(arg,'__iter__')` because `is_list_like` already checks for that